webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultGroupService.java

```diff
@@ -28,8 +30,15 @@
     }
 
     @Override
-    public Set<Group> getByQuery(final User user, final GroupQuery query) {
-        return ImmutableSet.copyOf(repository.findAllByQuery(user.getPermissionsById().get("GROUP_READ").getScope(), query));
+    public GroupSet getByQuery(final User user, final GroupQuery query) {
+        final Permission writePermission = user.getPermissionsById().get("GROUP_WRITE");
+        final Permission readPermission = user.getPermissionsById().get("GROUP_READ");
+
+        return GroupSet
+                .builder()
+                .groups(ImmutableSet.copyOf(repository.findAllByQuery(readPermission.getScope(), query))                )
+                .writeable(writePermission != null && schoolService.existsByPermissionScope(writePermission.getScope(), query.getSchoolId()))
```

Comment:
It's possible for a user to have GROUP_READ for a group but not have GROUP_WRITE, so we need to make this check agains the given schoolId.